### PR TITLE
Started to implement cached auth tokens

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #57 
Started to implement cached auth tokens, will continue in the next PRs.

### CODE REVIEW

File: src/main/java/com/artipie/conda/CachedAuthTokens.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+/**
+ * Cached authentication tokens.
+ * @since 0.5
+ */
+public final class CachedAuthTokens implements AuthTokens {
+
+    /**
+     * One hour in millis.
+     */
+    private static final int HOUR = 60 * 1000;
```
Comment (g4s8) [inline src/main/java/com/artipie/conda/CachedAuthTokens.java:25]: It's not an hour - it's a minute :)

File: src/main/java/com/artipie/conda/CachedAuthTokens.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    /**
+     * Ctor.
+     * @param origin Origin AuthTokens
+     */
+    public CachedAuthTokens(final AuthTokens origin) {
+        this(
+            CacheBuilder.newBuilder()
+                .expireAfterAccess(CachedAuthTokens.HOUR, TimeUnit.MILLISECONDS)
```
Comment (g4s8) [inline src/main/java/com/artipie/conda/CachedAuthTokens.java:54]: Maybe it'll be clearer to use `expireAfterAccess(1, TimeUnit.HOUR)`?

File: src/main/java/com/artipie/conda/CachedAuthTokens.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                }
+            );
+        } else if (!item.expired()) {
+            res = CompletableFuture.completedFuture(Optional.of(item));
+        }
+        return res;
+    }
+
+    @Override
+    @SuppressWarnings("PMD.ConfusingTernary")
```
Comment (g4s8) [inline src/main/java/com/artipie/conda/CachedAuthTokens.java:79]: I don't see any ternary in this method.

File: src/main/java/com/artipie/conda/CachedAuthTokens.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        if (!token.isPresent()) {
+            res = this.origin.find(username).thenApply(
+                tkn -> {
+                    tkn.ifPresent(present -> this.cache.put(present.token(), present));
+                    return tkn;
+                }
+            );
+        } else if (!token.get().expired()) {
+            res = CompletableFuture.completedFuture(token);
+        }
```
Comment (g4s8) [inline src/main/java/com/artipie/conda/CachedAuthTokens.java:94]: It looks like the rest of both functions is very similar. Maybe we can refactor it?

**Review by g4s8 — CHANGES_REQUESTED** [review]
@olenagerasimova please check my comments

File: src/main/java/com/artipie/conda/CachedAuthTokens.java (hunk 3 above)
Comment (olenagerasimova) [reply src/main/java/com/artipie/conda/CachedAuthTokens.java]: Neither do I, but qulice is complaning.

**Comment by olenagerasimova** [comment]
@g4s8 thanks, fixed and answered

**Comment by olenagerasimova** [comment]
@g4s8 please, check/approve

**Review by g4s8 — APPROVED** [review]
@olenagerasimova thanks

**Pull request merged**